src/main/scala/viper/gobra/frontend/Desugar.scala

```diff
@@ -164,21 +165,21 @@ object Desugar {
 
     private val nm = new NameManager
 
-    type Identity = Meta
+    type Identity = (Meta, TypeInfo)
 
-    private def abstraction(id: PIdnNode): Identity = {
+    private def abstraction(id: PIdnNode, info: TypeInfo = info): Identity = {
```

Comment:
If the reason to pass the `info` object as a parameter is that we are running into `node not in tree exception`, then I would not use a default value for this parameter, as doing so might still hide some bugs from before where we pass the wrong info. Having to think explicitly about every `info` that we pass, while annoying, is probably safer.

Reply:
Agreed, I have removed the default value from all functions that had it